src/background/services/paymentSession.ts

```diff
@@ -336,4 +274,57 @@ export class PaymentSession {
     this.amount = amount.toString()
     this.intervalInMs = Number((amount * BigInt(HOUR_MS)) / BigInt(this.rate))
   }
+
+  private async payContinuous() {
+    try {
+      const outgoingPayment =
+        await this.openPaymentsService.createOutgoingPayment({
+          walletAddress: this.sender,
+          incomingPaymentId: this.incomingPaymentUrl,
+          amount: this.amount
+        })
+      const { receiveAmount, receiver: incomingPayment } = outgoingPayment
+      const monetizationEventDetails: MonetizationEventDetails = {
+        amountSent: {
+          currency: receiveAmount.assetCode,
+          value: transformBalance(receiveAmount.value, receiveAmount.assetScale)
+        },
+        incomingPayment,
+        paymentPointer: this.receiver.id
+      }
+
+      sendMonetizationEvent({
+        tabId: this.tabId,
+        frameId: this.frameId,
+        payload: {
+          requestId: this.requestId,
+          details: monetizationEventDetails
+        }
+      })
+
+      // TO DO: find a better source of truth for deciding if overpaying is applicable
+      if (this.intervalInMs > 1000) {
+        this.tabState.saveOverpaying(this.tab, this.url, {
+          walletAddressId: this.receiver.id,
+          monetizationEvent: monetizationEventDetails,
+          intervalInMs: this.intervalInMs
+        })
+      }
```

Comment:
Should we move these out of `try catch {}`? Just to be clear that these can't throw?

Reply:
for me this doesn't particulary make sense to move them out, since they are dependent on the outgoingPayment being received successfully

Reply:
That I understand, but just meant as code style.
```ts
const outgoingPayment = await create(...).catch(error => {...})

if (!outgoingPayment) return

sendMonetizationEvent()
...
```